src/Core/Logger.php

```diff
@@ -67,73 +65,22 @@ class Logger extends BaseObject
 
 	/**
 	 * Sets the default logging handler for Friendica.
-	 * @todo Can be combined with other handlers too if necessary, could be configurable.
 	 *
 	 * @param LoggerInterface $logger The Logger instance of this Application
-	 *
-	 * @throws InternalServerErrorException if the logger factory is incompatible to this logger
 	 */
-	public static function setLogger($logger)
+	public static function init($logger)
 	{
-		$debugging = Config::get('system', 'debugging');
-		$logfile = Config::get('system', 'logfile');
-		$loglevel = Config::get('system', 'loglevel');
-
-		if (!$debugging || !$logfile) {
-			return;
-		}
-
-		$loglevel = self::mapLegacyConfigDebugLevel((string)$loglevel);
-
-		LoggerFactory::addStreamHandler($logger, $logfile, $loglevel);
-
 		self::$logger = $logger;
-
-		$logfile = Config::get('system', 'dlogfile');
-
-		if (!$logfile) {
-			return;
-		}
-
-		$developIp = Config::get('system', 'dlogip');
-
-		self::$devLogger = LoggerFactory::createDev('develop', $developIp);
-		LoggerFactory::addStreamHandler(self::$devLogger, $logfile, LogLevel::DEBUG);
 	}
 
 	/**
-	 * Mapping a legacy level to the PSR-3 compliant levels
-	 * @see https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md#5-psrlogloglevel
-	 *
-	 * @param string $level the level to be mapped
+	 * Sets the default dev-logging handler for Friendica.
 	 *
-	 * @return string the PSR-3 compliant level
+	 * @param LoggerInterface $logger The Logger instance of this Application
 	 */
-	private static function mapLegacyConfigDebugLevel($level)
+	public static function setDevLogger($logger)
```

Comment:
Please add the `LoggerInterface` type-hint.
```suggestion
	public static function setDevLogger(LoggerInterface $logger)
```